common/lib/xmodule/xmodule/js/src/video/09_completion.js

```diff
@@ -0,0 +1,123 @@
+(function(define) {
+    'use strict';
+    define('video/09_completion.js', [], function() {
+    /**
+     * Completion handler
+     * @exports video/09_completion.js
+     * @constructor
+     * @param {Object} state The object containing the state of the video
+     * @return {jquery Promise}
+     */
+        var VideoCompletionHandler = function(state) {
+            if (!(this instanceof VideoCompletionHandler)) {
+                return new VideoCompletionHandler(state);
+            }
+            this.state = state;
+            this.state.completionHandler = this;
+            this.initialize();
+            return $.Deferred().resolve().promise();
+        };
+
+        VideoCompletionHandler.prototype = {
+            destroy: function() {
+                this.el.remove();
+                this.state.el.off('destroy', this.destroy);
+                delete this.state.completionHandler;
+            },
+
+        /** Initializes the module. */
+            initialize: function() {
+                this.lastSentAt = undefined;
+                this.complete = false;
+                this.completionPercentage = this.state.config.completionPercentage;
+                this.startTime = this.state.config.startTime;
+                this.endTime = this.state.config.endTime;
+                if (this.endTime) {
+                    this.completeAfter = this.startTime + (this.endTime - this.startTime) * this.completionPercentage;
+                }
+                this.bindHandlers();
+            },
+
+        /** Bind any necessary function callbacks to DOM events. */
+            bindHandlers: function() {
+                var self = this;
+
+            /** Event handler to check if the video is complete, and submit a completion if it is.
+             *
+             *  If the timeupdate handler doesn't fire after the required percentage, this will catch
+             *  any fully complete videos.
+             **/
+                this.state.el.on('ended.completion', function() {
+                    if (self.complete) {
+                        return;
+                    }
+                    self.markCompletion();
+                });
+
+            /** Event handler to check video progress, and mark complete if
+             *  greater than completionPercentage
+             **/
+                this.state.el.on('timeupdate.completion', function(ev, currentTime) {
+                    var duration;
+                    if (self.complete) {
+                        return;
+                    }
+                    if (self.lastSentAt !== undefined && currentTime - self.lastSentAt < 3.0) {
```

Comment:
Could you pull that `3.0` into a variable at the top, so we can easily adjust if needed?